### DISCUSSION TITLE
How to estimate memory usage of a loaded image?

### DISCUSSION THREAD

**Discussion by urain39** [ImageMagick/ImageMagick#7629] [Help]
### ImageMagick version

7.1.1-38 Q16

### Operating system, version and so on

Android 14 (Termux)

### Description

Currently, I use `width*height*channels*quantum*2.6` to calculate the maximum memory usage, but this is only relatively accurate when loading PNG images. For JPG images, the factor of 2.6 may need to be replaced with 1.6.

Is there a more accurate and universal method?

**Comment by snibgo** [comment]
When an image is in memory, each pixel needs space, and there is an overhead per image.

If you have HDRI, each pixel needs 4 bytes per channel. If you don't have HDRI, then Q8 needs 1 byte per channel; Q16 needs 2 bytes per channel, Q32 needs 4 bytes per channel.

The overhead per image is about 26500 bytes.

When processing is done (eg blurring, writing image to file), extra memory is needed while that operation is being performed.
